src/devices/bus/nes_ctrl/powerpad.cpp

```diff
@@ -94,43 +95,41 @@ nes_powerpad_device::nes_powerpad_device(const machine_config &mconfig, const ch
 void nes_powerpad_device::device_start()
 {
 	save_item(NAME(m_latch));
-}
-
-
-//-------------------------------------------------
-//  device_reset
-//-------------------------------------------------
-
-void nes_powerpad_device::device_reset()
-{
-	m_latch[0] = 0;
-	m_latch[1] = 0;
+	save_item(NAME(m_strobe));
 }
 
 
 //-------------------------------------------------
 //  read
 //-------------------------------------------------
 
-uint8_t nes_powerpad_device::read_bit34()
+u8 nes_powerpad_device::read_bit34()
 {
-	uint8_t ret = 0;
-	ret |= (m_latch[0] & 0x01) << 3;
+	if (BIT(m_strobe, 0))
+	{
+		m_latch[0] = m_ipt1->read();
+		m_latch[1] = m_ipt2->read() | 0xf0;
```

Comment:
Why not just use `IPT_ACTIVE_LOW` for the unused input bits so it returns the value you want and you don’t need to or it with a constant here?

Reply:
Thanks for the tip. Makes things slightly cleaner. I did that and also went ahead and made m_ipt an array.